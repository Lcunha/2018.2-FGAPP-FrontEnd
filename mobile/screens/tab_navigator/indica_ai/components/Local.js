import React, { Component}   from 'react';
import {
  View,
  Text,
  StyleSheet
} from 'react-native';
import { Card, CardItem } from 'native-base' 


export default class Local extends Component {
  constructor(props) {
    super(props);
  }
  render() {

    const name = this.props.name;
    const description = this.props.description;
  
    return(
        <Card style={styles.localCard}> 
        
         <CardItem header bordered>
            <Text style={styles.localName}>
             {name}
           </Text>
         </CardItem>

         <CardItem>
            <Text style={styles.localDescription}>
              {description}
           </Text>
         </CardItem>

        </Card>
    );
  }
}


const styles = StyleSheet.create({

  localCard: {
      borderRadius: 7,
  },

  localDescription: {
    marginLeft: 0,
    fontSize: 15
  },
  localName: {
    fontWeight: 'bold',
    color: '#333',
    fontSize: 20
  },

});
