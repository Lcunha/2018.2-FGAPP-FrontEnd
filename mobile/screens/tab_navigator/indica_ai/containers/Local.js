import React from 'react';
import {
  View,
  Text,
  StyleSheet
} from 'react-native';

export default class Local extends React.Component {
  render() {
    return(
      <View style={styles.local}>
        <View style={styles.localInfo}>
          <Text style={styles.localName}>
            {this.props.data.name}
          </Text>
        </View>
      </View>
    );
  }
}


const styles = StyleSheet.create({

  local: {
    padding: 15,
    borderWidth: 1,
    borderColor: '#333',
    height: 120,
    borderRadius: 5,
    marginTop: 20,
  },
  localInfo:{
    marginLeft: 0,
  },
  localName: {
    fontWeight: 'bold',
    color: '#333'
  },

});
