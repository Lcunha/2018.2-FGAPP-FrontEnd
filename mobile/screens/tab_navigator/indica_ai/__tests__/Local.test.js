import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

import Local from '../components/Local';

Enzyme.configure({adapter: new Adapter()});

  it('renders correctly', () => {
    const tree = renderer.create(<Local />).toJSON();
    expect(tree).toMatchSnapshot();
});
