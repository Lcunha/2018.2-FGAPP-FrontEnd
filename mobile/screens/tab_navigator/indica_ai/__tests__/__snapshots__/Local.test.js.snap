// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "borderRadius": 7,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      bordered={true}
      header={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "marginLeft": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
